homeassistant/components/shelly/bluetooth/__init__.py

```diff
@@ -0,0 +1,97 @@
+"""Bluetooth support for shelly."""
+from __future__ import annotations
+
+from typing import TypedDict
+
+from aioshelly.rpc_device import RpcDevice
+
+from homeassistant.components.bluetooth import (
+    HaBluetoothConnector,
+    async_get_advertisement_callback,
+    async_register_scanner,
+)
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant, callback as hass_callback
+from homeassistant.helpers.device_registry import format_mac
+
+from ..coordinator import ShellyRpcCoordinator
+from .scanner import ShellyBLEScanner
+
+BLE_SCRIPT_NAME = "homeassistant_ble_integration"
+
+BLE_CODE = """
+// Home Assistant BLE script v0.1.0
+BLE.Scanner.Subscribe(function (ev, res) {
+    if (ev === BLE.Scanner.SCAN_RESULT) {
+        Shelly.emitEvent("ble.scan_result", [
+            res.addr,
```

Comment:
Maybe put a leading entry here for a "format version", in case it needs to change later

Reply:
Will probably make the version the ha version so we know where it came from

Will also make the event formatted as the same const

Reply:
b67a146ac574c104817548786eeb2968be445fe4